test/test_check.cc

```diff
@@ -0,0 +1,177 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <sstream>
+#include <vector>
+
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+
+#include "tc/core/check.h"
+
+// XXX:gtest doesn't define a macro that inspects the contents of the exception
+#define ASSERT_THROW_WHAT(x, y)      \
+  try {                              \
+    (x);                             \
+    ASSERT_TRUE(false);              \
+  } catch (std::runtime_error & e) { \
+    ASSERT_EQ(y, e.what());          \
+  }
+
+TEST(CHECK, Plain) {
+  ASSERT_NO_THROW(TC_CHECK(true));
+  {
+    std::stringstream expected;
+    expected << "Check failed [" << __FILE__ << ':' << __LINE__ + 1 << ']';
+    ASSERT_THROW_WHAT(TC_CHECK(false), expected.str());
+  }
+
+  {
+    std::stringstream expected;
+    expected << "Check failed [" << __FILE__ << ':' << __LINE__ + 2 << ']'
+             << ": 1+1=3";
+    ASSERT_THROW_WHAT(TC_CHECK(false) << "1+1=3", expected.str());
+  }
+}
+
+TEST(CHECK, Vector) {
+  std::stringstream expected;
+  expected << "Check failed [" << __FILE__ << ':' << __LINE__ + 3 << ']'
+           << ": 1,2,3,4";
+  ASSERT_THROW_WHAT(
+      (TC_CHECK(false) << std::vector<int>{1, 2, 3, 4}), expected.str());
+}
+
+TEST(CHECK, EQ) {
+  ASSERT_NO_THROW(TC_CHECK_EQ(1, 1));
+  ASSERT_NO_THROW(TC_CHECK_EQ(std::string("aaa"), std::string("aaa")));
+  {
+    std::stringstream expected;
+    expected << "Check failed [" << __FILE__ << ':' << __LINE__ + 2 << "] "
+             << "1 not equal to 2";
+    ASSERT_THROW_WHAT(TC_CHECK_EQ(1, 2), expected.str());
+  }
+
+  {
+    std::stringstream expected;
+    expected << "Check failed [" << __FILE__ << ':' << __LINE__ + 5
+             << "] 1 not equal to 2: 2+2=5";
+    ASSERT_THROW_WHAT(
+        TC_CHECK_EQ(1, 2) << "2+2"
+                          << "=5",
+        expected.str());
+  }
+}
+
+TEST(CHECK, NE) {
+  ASSERT_NO_THROW(TC_CHECK_NE(1, 2));
+  ASSERT_NO_THROW(TC_CHECK_NE(std::string("aaa"), std::string("baa")));
+  {
+    std::stringstream expected;
+    expected << "Check failed [" << __FILE__ << ':' << __LINE__ + 2
+             << "] 1 equal to 1";
+    ASSERT_THROW_WHAT(TC_CHECK_NE(1, 1), expected.str());
+  }
+
+  {
+    std::stringstream expected;
+    expected << "Check failed [" << __FILE__ << ':' << __LINE__ + 2
+             << "] 1 equal to 1: 2+2=5";
+    ASSERT_THROW_WHAT(TC_CHECK_NE(1, 1) << "2+2=5", expected.str());
+  }
+}
+
+TEST(CHECK, LT) {
+  ASSERT_NO_THROW(TC_CHECK_LT(1, 2));
+  ASSERT_NO_THROW(TC_CHECK_LT(std::string("aaa"), std::string("baa")));
+  {
+    std::stringstream expected;
+    expected << "Check failed [" << __FILE__ << ':' << __LINE__ + 2
+             << "] 1 not less than " << 1;
```

Comment:
Why is the second "1" outside the string (and not the first one)?
Similarly below.

Reply:
No reason. I'll fix it.